### PR TITLE
Draft Release automation

### PR DESCRIPTION
**Pull request by wtrocki**
## Description

This PR creates release notes and draft releases in github. 
We will start with a draft release first and then have separate PR to set the release's property to false once we have the rest of the automation in place.

Follow-up PRs will introduce multi-version support, s3 automation etc.

## Verification

CI/CD runs with success. The release is created with breaking changes included.
https://github.com/mongodb/atlas-sdk-go/actions/runs/5673051560/job/15373647341?pr=129

### CODE REVIEW

File: .github/workflows/autorelease.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: Autorelease
+on:
+  push:
+    branches:
+      - main
+      - release-automation
```
Comment (wtrocki) [inline .github/workflows/autorelease.yaml:6]: Will be removed after review. This is only to test job. 
Job runs right now and created valid release.
```suggestion
```

File: BREAKING_CHANGES.md [hunk 2]
```diff
@@ -1,20 +0,0 @@
-## v20230202003
```
Comment (wtrocki) [inline BREAKING_CHANGES.md:1]: We split this file into changes per release for clarity.

File: tools/releaser/breaking_changes/v20230201002.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+## Breaking Changes
```
Comment (andreaangiolillo) [inline tools/releaser/breaking_changes/v20230201002.md:1]: How is this file generated for a new release? Is this a manual process?

File: .github/workflows/autorelease.yaml (hunk 1 above)
Comment (andreaangiolillo) [inline .github/workflows/autorelease.yaml:6]: what changes are inside in release-automation that are not in the main branch?

File: .github/workflows/autorelease.yaml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+name: Autorelease
+on:
```
Comment (andreaangiolillo) [inline .github/workflows/autorelease.yaml:2]: I would add   `workflow_dispatch: {}` to be able to run the action manually -> easier to test so you don't have to merge a commit into the branch (feel free to ignore)
```suggestion
on:
  workflow_dispatch: {} # workflow can be run manually
```

**Review by andreaangiolillo — APPROVED** [review]
LGTM

File: .github/workflows/autorelease.yaml (hunk 1 above)
Comment (wtrocki) [reply .github/workflows/autorelease.yaml:6]: This is only to enable verification of the PR. Removing now

File: .github/workflows/autorelease.yaml (hunk 4 above)
Comment (wtrocki) [reply .github/workflows/autorelease.yaml:2]: Amazing point. Will add that as well but it needs some docs

File: tools/releaser/breaking_changes/v20230201002.md (hunk 3 above)
Comment (wtrocki) [reply tools/releaser/breaking_changes/v20230201002.md:1]: Yes. Establishing breaking changes is the only manual process. 
It is based on exemptions in openapi.

**Pull request merged**